packages/@lwc/template-compiler/src/__tests__/fixtures/regression/handler-memoization/expected.js

```diff
@@ -22,9 +22,11 @@ function tmpl($api, $cmp, $slotset, $ctx) {
       api_static_part(
         0,
         {
-          on: {
-            click: _m1 || ($ctx._m1 = api_bind($cmp.create)),
-          },
+          on:
+            _m1 ||
+            ($ctx._m1 = {
+              click: api_bind($cmp.create),
+            }),
```

Comment:
This looks good now that task.delete is not memoized, but I think we should also test multiple handlers of different scopes on the same element inside a loop.

I'm not sure whether to create a new fixture or edit this one to add a $cmp scoped listener to the [X] button.

I guess a karma test would be nice if it's possible to reproduce the behavior mentioned here (it could be added to #4480): https://github.com/salesforce/lwc/pull/4468/files#r1728035122

> Upon reflection, the above would be a breaking change. If an element has mixed event listeners accessing different scopes, e.g.:
```html
<button onclick={task.delete} ontouchstart={foo}></button>
```
> ... then the foo listener would get re-bound on every render, potentially changing the behavior in an observable way.

Reply:
The fixture tests are good for a quick sniff check, but I'd really prefer a Karma test here to confirm that the behavior remains the same, yeah.

For the above `packages/@lwc/integration-karma/test/events/memoization/index.spec.js`, we could add a third test with a mix of 1) local and 2) non-local event listeners on the same element, and then confirm that they behave differently.

Reply:
Added the test (x/mixed). I believe it's correct, but feel free to tweak it otherwise.